### PR TITLE
[MIG] stock_reserve_sale: Migration to 16.0

### PR DESCRIPTION
**Pull request by XanderDeJaegere**
[[MIG]stock_reserve_sale: migrate from v13.0 to v16.0](https://github.com/OCA/stock-logistics-warehouse/pull/1695/commits/4ff4e11ca24bf32bbf95a92517a175b7fa0f112a)

### CODE REVIEW

**Comment by rvalyi** [comment]
Hello, you might consider rebuilding the migration from this v14 migration PR as it brings more bugfix that is not present here https://github.com/OCA/stock-logistics-warehouse/pull/1696

**Comment by hugo-cordoba** [comment]
Good morning @rousseldenis , could you add this migration to the migration task and add the milestone? Thanks!! 

Regards.

**Comment by rvalyi** [comment]
/ocabot migration stock_reserve_sale

@HugoCordobaLeal still you should really look at the v14 PR I mentioned and possibly cherry-pick the v14 migration commit as it contains important cleanup that is missing here.

**Comment by amarcosg** [comment]
Hi @XanderDeJaegere do you have any plans to include the changes proposed here https://github.com/OCA/stock-logistics-warehouse/pull/1695#issuecomment-1521339035?

Thanks!

**Comment by XanderDeJaegere** [comment]
I will take a look on Saturday @amarcosg . My apologies, I don't visit this account often.

**Comment by amarcosg** [comment]
No problem. Thank you so much @XanderDeJaegere

**Comment by XanderDeJaegere** [comment]
I haven't forgotten about this, but there were some personal events that happened

File: stock_reserve_sale/model/sale.py [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+    is_stock_reservable = fields.Boolean(
+        compute="_compute_stock_reservation",
+        readonly=True,
+        store=True,
+        string="Can Have Stock Reservations",
+    )
+
+    def release_all_stock_reservation(self):
+        line_ids = [line.id for order in self for line in order.order_line]
+        lines = self.env["sale.order.line"].browse(line_ids)
```
Comment (ferran-S73) [inline stock_reserve_sale/model/sale.py:45]:
```suggestion
        lines = self.order_line
```

**Comment by hugo-cordoba** [comment]
Hi @XanderDeJaegere! Are you going to go ahead with this or can I create a new PR?

**Comment by rousseldenis** [comment]
@XanderDeJaegere Do you agree we switch to #1788 ?

**Comment by rousseldenis** [comment]
> @XanderDeJaegere Do you agree we switch to #1788 ?

@XanderDeJaegere

**Comment by XanderDeJaegere** [comment]
Yes you can switch to #1788 , the cherry picking I had to do never felt resolved so I didn't know what I could do for it.

**Pull request closed without merging**